src/gui/torrentcontentmodel.cpp

```diff
@@ -81,44 +86,80 @@ namespace
             return cached;
         }
     };
-#if defined(Q_OS_WIN)
-    // See QTBUG-25319 for explanation why this is required
-    class WinShellFileIconProvider: public UnifiedFileIconProvider
+
+#ifdef QBT_PIXMAP_CACHE_FOR_FILE_ICONS
+    Q_DECL_UNUSED struct PixmapCacheSetup
+    {
+        static const int PixmapCacheForIconsSize = 2 * 1024 * 1024; // 2 MiB for file icons
+
+        PixmapCacheSetup()
+        {
+            QPixmapCache::setCacheLimit(QPixmapCache::cacheLimit() + PixmapCacheForIconsSize);
+        }
+
+        ~PixmapCacheSetup()
+        {
+            Q_ASSERT(QPixmapCache::cacheLimit() > PixmapCacheForIconsSize);
+            QPixmapCache::setCacheLimit(QPixmapCache::cacheLimit() - PixmapCacheForIconsSize);
+        }
+    };
+
+    PixmapCacheSetup pixmapCacheSetup;
+
+    class CachingFileIconProvider: public UnifiedFileIconProvider
     {
     public:
         using QFileIconProvider::icon;
 
-        QIcon icon(const QFileInfo &info) const override
+        QIcon icon(const QFileInfo &info) const final
```

Comment:
What a coincidence. StackOverflow has the same question: https://stackoverflow.com/questions/29412412/does-final-imply-override